common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt

```diff
@@ -829,7 +829,7 @@ fun <T> Entity<T>.isActive() = when {
     (domain == "cover") -> state != "closed"
     (domain in listOf("device_tracker", "person")) -> state != "not_home"
     (domain == "lawn_mower") -> state in listOf("mowing", "error")
-    (domain == "lock") -> state != "locked"
+    (domain == "lock") -> state == "locked"
```

Comment:
After having some discussion and another user mentioning this in Discord we have noticed that the lock icon which is controlled by Android actually matches your suggested behavior.

Can you please add a comment to this line mentioning that we are deviating from the frontend to better match Android native behavior? This way we can remember that we need to keep this logic.

![image](https://github.com/home-assistant/android/assets/1634145/3912ddba-2739-4a2b-be89-b2ad425f7e60)

https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/controls/LockControl.kt#L40